### ISSUE TITLE
remote dependencies in profile dcf file

### ISSUE THREAD

**Issue by yonicd** [rstudio/renv#904]
Continuing from [original post](https://community.rstudio.com/t/how-to-set-up-profiles-in-renv/125538/5) on rs community

if there are remote dependencies in the profile Config then the regex [pattern](https://github.com/rstudio/renv/blob/main/R/dependencies.R#L455_L458) doesn't take into account the option for a remote package install

eg `Config/renv/profiles/trial/dependencies: r-lib/devtools`

``` r
* Initializing project ...
* Discovering package dependencies ... Done!
* Copying packages into the cache ... Done!
* Resolving missing dependencies  ... 
Installing progress [1.2.2] ...
	OK [linked cache]
The following package(s) were not installed successfully:

	[r]: package 'r' is not available

You may need to manually download and install these packages.
```

if i pass the `parts` in the lapply through `renv_remotes_resolve` then i get the right parsing of the dependency

```r
[[1]]
[[1]]$Package
[1] "devtools"

[[1]]$Version
[1] "2.4.3.9000"

[[1]]$Source
[1] "GitHub"

[[1]]$RemoteType
[1] "github"

[[1]]$RemoteHost
[1] "api.github.com"

[[1]]$RemoteUsername
[1] "r-lib"

[[1]]$RemoteRepo
[1] "devtools"

[[1]]$RemoteRef
[1] "main"

[[1]]$RemoteSha
[1] "a999c84c03a8c955bf1bc81b2ae3f6d8e1c7110b"
```

could information from that be used to replace the contents of `data`

```r
data
[[1]]
list()

[[2]]
list()

[[3]]
list()

[[4]]
list()

[[5]]
                           Source Package Require Version   Dev
1 C:/projects/newtest/DESCRIPTION       r                 FALSE
```

**Comment by kevinushey** [comment]
Thanks for the bug report -- indeed, arbitrary remotes are not yet supported here. I'll have to think about what the right way to support this is.

**Comment by hadley** [comment]
Is it ok to handle as remotes elsewhere? i.e. you supply the package name in `Config/renv/profiles/trial/dependencies` and then the specific package source in `Remotes`?  This would mean that you can't specify different remotes in different profiles, but that feels like a fairly exotic need.

**Comment by hadley** [comment]
Fixed in 4343e7cfefa48d70e856bdfc16d69169fec7d64b
